### PR TITLE
nodebuilder: make proxy constructors private in all modules

### LINKED ISSUE #1375: nodebuilder: Private proxy constructors in all modules
In our modules, we create proxy constructors and keep them public quite often, but they should not be so, and this task it reviews all modules for such and make them private.

### PR DESCRIPTION
**Pull request by Jpserrat**
## Overview
Closes #1375 
Made all the proxy constructors private except for the Key constructor of the p2p package cause it is also used in swamp

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by Wondertan — CHANGES_REQUESTED** [review]
There are many comments with decapitalized names, pls keep them capital

File: nodebuilder/p2p/bitswap.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ const (
 )
 
 // DataExchange provides a constructor for IPFS block's DataExchange over BitSwap.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/bitswap.go:27]:
```suggestion
// dataExchange provides a constructor for IPFS block's DataExchange over BitSwap.
```

File: nodebuilder/p2p/host.go [hunk 2]
```diff
@@ -23,12 +23,12 @@ import (
 
 // RoutedHost constructs a wrapped Host that may fallback to address discovery,
 // if any top-level operation on the Host is provided with PeerID(Hash(PbK)) only.
-func RoutedHost(base HostBase, r routing.PeerRouting) host.Host {
+func routedHost(base HostBase, r routing.PeerRouting) hst.Host {
 	return routedhost.Wrap(base, r)
 }
 
 // Host returns constructor for Host.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/host.go:30]:
```suggestion
// host returns constructor for Host.
```

File: nodebuilder/p2p/ipld.go [hunk 3]
```diff
@@ -7,6 +7,6 @@ import (
 )
 
 // BlockService constructs IPFS's BlockService for fetching arbitrary Merkle structures.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/ipld.go:9]:
```suggestion
// blockService constructs IPFS's BlockService for fetching arbitrary Merkle structures.
```

File: nodebuilder/p2p/misc.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ import (
 )
 
 // ConnManagerConfig configures connection manager.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/misc.go:14]:
```suggestion
// connManagerConfig configures connection manager.
```

File: nodebuilder/p2p/misc.go [hunk 5]
```diff
@@ -21,16 +21,16 @@ type ConnManagerConfig struct {
 }
 
 // DefaultConnManagerConfig returns defaults for ConnManagerConfig.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/misc.go:23]:
```suggestion
// defaultConnManagerConfig returns defaults for ConnManagerConfig.
```

File: nodebuilder/p2p/misc.go [hunk 6]
```diff
@@ -21,16 +21,16 @@ type ConnManagerConfig struct {
...
-	return ConnManagerConfig{
+func defaultConnManagerConfig() connManagerConfig {
+	return connManagerConfig{
 		Low:         50,
 		High:        100,
 		GracePeriod: time.Minute,
 	}
 }
 
 // ConnectionManager provides a constructor for ConnectionManager.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/misc.go:32]:
```suggestion
// connectionManager provides a constructor for ConnectionManager.
```

File: nodebuilder/p2p/misc.go [hunk 7]
```diff
@@ -54,11 +54,11 @@ func ConnectionManager(cfg Config, bpeers Bootstrappers) (connmgri.ConnManager,
 }
 
 // ConnectionGater constructs a BasicConnectionGater.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/misc.go:56]:
```suggestion
// connectionGater constructs a BasicConnectionGater.
```

File: nodebuilder/p2p/misc.go [hunk 8]
```diff
@@ -54,11 +54,11 @@ func ConnectionManager(cfg Config, bpeers Bootstrappers) (connmgri.ConnManager,
...
-func ConnectionGater(ds datastore.Batching) (*conngater.BasicConnectionGater, error) {
+func connectionGater(ds datastore.Batching) (*conngater.BasicConnectionGater, error) {
 	return conngater.NewBasicConnectionGater(ds)
 }
 
 // PeerStore constructs a PeerStore.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/misc.go:61]:
```suggestion
// peerStore constructs a PeerStore.
```

File: nodebuilder/p2p/pubsub.go [hunk 9]
```diff
@@ -5,20 +5,20 @@ import (
...
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	pubsub_pb "github.com/libp2p/go-libp2p-pubsub/pb"
-	"github.com/libp2p/go-libp2p/core/host"
+	hst "github.com/libp2p/go-libp2p/core/host"
 	"github.com/libp2p/go-libp2p/core/protocol"
 	"go.uber.org/fx"
 	"golang.org/x/crypto/blake2b"
 )
 
 // PubSub provides a constructor for PubSub protocol with GossipSub routing.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/pubsub.go:14]:
```suggestion
// pubSub provides a constructor for PubSub protocol with GossipSub routing.
```

File: nodebuilder/p2p/routing.go [hunk 10]
```diff
@@ -13,13 +13,13 @@ import (
 
 // ContentRouting constructs nil content routing,
```
Comment (distractedm1nd) [inline nodebuilder/p2p/routing.go:14]:
```suggestion
// contentRouting constructs nil content routing,
```

File: nodebuilder/p2p/routing.go [hunk 11]
```diff
@@ -13,13 +13,13 @@ import (
...
 // as for our use-case existing ContentRouting mechanisms, e.g DHT, are unsuitable
-func ContentRouting(r routing.PeerRouting) routing.ContentRouting {
+func contentRouting(r routing.PeerRouting) routing.ContentRouting {
 	return r.(*dht.IpfsDHT)
 }
 
 // PeerRouting provides constructor for PeerRouting over DHT.
```
Comment (distractedm1nd) [inline nodebuilder/p2p/routing.go:20]:
```suggestion
// peerRouting provides constructor for PeerRouting over DHT.
```

File: nodebuilder/rpc/constructors.go [hunk 12]
```diff
@@ -14,7 +14,7 @@ import (
 )
 
 // RegisterEndpoints registers the given services on the rpc.
```
Comment (distractedm1nd) [inline nodebuilder/rpc/constructors.go:16]:
```suggestion
// registerEndpoints registers the given services on the rpc.
```

File: nodebuilder/state/core.go [hunk 13]
```diff
@@ -10,7 +10,7 @@ import (
 
 // CoreAccessor constructs a new instance of state.Module over
```
Comment (distractedm1nd) [inline nodebuilder/state/core.go:11]:
```suggestion
// coreAccessor constructs a new instance of state.Module over
```

**Review by distractedm1nd — CHANGES_REQUESTED** [review]
Talked with @Wondertan - the godoc comments were right, the concern was the rest - so I've made suggestions to these to make it easy to apply them

Thank you for this! Your other PR is also in our pipeline :)

**Review by distractedm1nd — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]
LGTM

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**